### PR TITLE
example project: fix analysis issues and upgrade to null safety

### PR DESCRIPTION
**Pull request by AlexHartford**
Might be good to wait until `badges` null safety release hits stable.

### CODE REVIEW

**Comment by Ascenio** [comment]
Nice. But could you remove the code comments?

File: example/lib/main_advance.dart [hunk 1]
```diff
@@ -13,17 +13,12 @@ class TabPage extends StatefulWidget {
...
 
   PageController controller = PageController();
 
-  List<Color> colors = [
-    Colors.purple,
-    Colors.pink,
-    Colors.amber[600],
-    Colors.teal
-  ];
+  List<Color?> colors = [Colors.purple, Colors.pink, Colors.amber[600], Colors.teal];
```
Comment (Ascenio) [inline example/lib/main_advance.dart:21]: I think it would be better to make it a `List<Color>`

File: example/lib/main_gallery.dart [hunk 2]
```diff
@@ -32,7 +32,7 @@ class _MyHomePageState extends State<MyHomePage> {
   int badge = 0;
 
   List<GButton> tabs = [];
-  List<Color> colors = [
+  List<Color?> colors = [
```
Comment (Ascenio) [inline example/lib/main_gallery.dart:35]: Change for `List<Color>`

File: example/lib/main_gallery.dart [hunk 3]
```diff
@@ -107,7 +104,7 @@ class _MyHomePageState extends State<MyHomePage> {
                           )
                         ],
                         selectedIndex: selectedIndex,
-                        onTabChange: (index) => (_tabChanged(index))),
+                        onTabChange: (index) => _tabChanged(index)),
```
Comment (Ascenio) [inline example/lib/main_gallery.dart:107]: Could simplify like:
```dart
onTabChange: _tabChanged,
```

File: example/lib/main_gallery.dart [hunk 4]
```diff
@@ -174,23 +161,22 @@ class _MyHomePageState extends State<MyHomePage> {
...
-                                    "https://sooxt98.space/content/images/size/w100/2019/01/profile.png")),
+                              radius: 12,
+                              backgroundImage: NetworkImage(
+                                  'https://sooxt98.space/content/images/size/w100/2019/01/profile.png'),
+                            ),
                           )
                         ],
                         selectedIndex: selectedIndex,
-                        onTabChange: (index) => (_tabChanged(index))),
+                        onTabChange: (index) => _tabChanged(index)),
```
Comment (Ascenio) [inline example/lib/main_gallery.dart:171]: Could simplify to:
```dart
onTabChange: _tabChanged,
```

File: example/lib/main_gallery.dart [hunk 5]
```diff
@@ -237,35 +222,33 @@ class _MyHomePageState extends State<MyHomePage> {
                             )
                           ],
                           selectedIndex: selectedIndex,
-                          onTabChange: (index) => (_tabChanged(index)))),
+                          onTabChange: (index) => _tabChanged(index))),
```
Comment (Ascenio) [inline example/lib/main_gallery.dart:225]: Could simplify to:
```dart
onTabChange: _tabChanged,
```

File: example/lib/main_gallery.dart [hunk 6]
```diff
@@ -277,12 +260,12 @@ class _MyHomePageState extends State<MyHomePage> {
...
                           GButton(
-                            icon: LineIcons.gear,
-                            text: 'Setting',
+                            icon: LineIcons.cog,
+                            text: 'Settings',
                           )
                         ],
                         selectedIndex: selectedIndex,
-                        onTabChange: (index) => (_tabChanged(index))),
+                        onTabChange: (index) => _tabChanged(index)),
```
Comment (Ascenio) [inline example/lib/main_gallery.dart:268]: Could simplify to:
```dart
onTabChange: _tabChanged,
```

**Comment by AlexHartford** [comment]
Addressed comments, let me know if there's anything else!

File: lib/src/gnav.dart [hunk 7]
```diff
@@ -33,7 +33,7 @@ class GNav extends StatefulWidget {
 
   final List<GButton> tabs;
   final int selectedIndex;
-  final Function onTabChange;
+  final Function(int) onTabChange;
```
Comment (Ascenio) [inline lib/src/gnav.dart:36]: Could you change the type for ValueChanged<int>?

Comment (AlexHartford) [reply lib/src/gnav.dart]: Done, good catch, thanks

**Review by Ascenio — APPROVED** [review]
Good job :smiley:

**Comment by AlexHartford** [comment]
Can probably just merge this, seems that the badges package is taking a long time to publish a stable version of their null safe library. Seeing as it's just impacting the example and there are no bugs in this example, should go ahead and merge to avoid future conflicts.

**Pull request merged**